### PR TITLE
Adds a digging tool to people's survival boxes on icebox

### PR DESCRIPTION
**Pull request by wesoda25**
## About The Pull Request

Part two of multi z qol. Falling into holes SUCKS, and now crewmembers have the ability to do something about it! 

Also touches some json stuff. I can't actually test if the PR works, aside from compiling it (so no errors/warnings) since json is broken, and even with the changes from #54603 it doesn't load properly. So DNM for now, I just wanted to get this out there before the end of the month since I think it is a needed change.

## Why It's Good For The Game

Basically what I said above. No one likes falling into a hole, since all that happens is they ask for help on comms and then go afk. This will allow for interesting battles for survival in icebox, fighting to make it back to the station against everything in your path.

## Changelog
:cl:
add: Nanotrasen has finally decided to add a type of pickax, the emergency excavation tool, to crewmembers survival boxes aboard their lunar outposts. Those lost to the depths of the ice moon: your deaths were not in vain!
/:cl:

### CODE REVIEW

File: code/modules/jobs/job_types/_job.dm [hunk 1]
```diff
@@ -249,6 +249,14 @@
 		else
 			back = backpack //Department backpack
 
+	//checks map config to see if it uses custom survival boxes
+	SSmapping.HACK_LoadMapConfig()
+	if(SSmapping.config.box_suffix)
+		var/box_suffix = SSmapping.config.box_suffix
+		var/box_type
+		box_type = "[box][box_suffix]"
+		box = text2path(box_type)
```
Comment (Jared-Fogle) [inline code/modules/jobs/job_types/_job.dm:258]: Can you assert that the path actually exists?

File: _maps/icebox.json [hunk 2]
```diff
@@ -46,6 +46,7 @@
 		}
 	],
 	"minetype": "none",
+	"box_suffix": "/icebox",
```
Comment (Jared-Fogle) [inline _maps/icebox.json:49]: I'm not confident I like this over an array of additional items. This is more flexible, I guess?

File: code/datums/map_config.dm [hunk 3]
```diff
@@ -24,6 +24,8 @@
 
 	var/minetype = "lavaland"
 
+	var/box_suffix  = null
```
Comment (Jared-Fogle) [inline code/datums/map_config.dm:27]:
```suggestion
	var/box_suffix = null
```

File: code/modules/mining/equipment/mining_tools.dm [hunk 4]
```diff
@@ -117,6 +117,20 @@
 	w_class = WEIGHT_CLASS_NORMAL
 	custom_materials = list(/datum/material/iron=12050) //metal needed for a crowbar and for a knife, why the FUCK does a knife cost 6 metal sheets while a crowbar costs 0.025 sheets? shit makes no sense fuck this
 
+/obj/item/pickaxe/emergency
```
Comment (Jared-Fogle) [inline code/modules/mining/equipment/mining_tools.dm:120]: There already exists `/obj/item/pickaxe/emergency`. It's the emergency disembarkation tool. `code/modules/shuttle/emergency.dm`.

Comment (Jared-Fogle) [reply code/modules/mining/equipment/mining_tools.dm:120]: SpacemanDMM should lint for duplicate definitions like this, honestly. Oh well.

**Comment by wesoda25** [comment]
I’ll take a look at all of that tomorrow. Also wanna work on sprites a bit more.

**Comment by JohnFulpWillard** [comment]
Would it be possible to just put a survival pen (the mining one from bepis) in their box, if the map is set to icebox?

File: _maps/icebox.json (hunk 2 above)
Comment (Cyberboss) [reply _maps/icebox.json:49]: Do this differently. Create a `global` job_changes entry and add the field `box` and enter the full path to the box obj.

File: code/datums/map_config.dm [hunk 5]
```diff
@@ -137,6 +139,9 @@
 	if ("minetype" in json)
 		minetype = json["minetype"]
 
+	if("box_suffix" in json)
+		box_suffix = json["box_suffix"]
```
Comment (Cyberboss) [inline code/datums/map_config.dm:143]: And existence of the type should be validated here.

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by MrDoomBringer** [comment]
I feel like this tool won't be particularly useful without knowing which direction the entrance to the station is
Maybe add a small pinpointer or other device suited for this purpose as well?

**Comment by wesoda25** [comment]
Ok, pretty busy so I’ll try the code stuff when I can. For now though 

> Would it be possible to just put a survival pen (the mining one from bepis) in their box, if the map is set to icebox?

I don’t see how this would solve the issue of people falling into holes with no way out. 


> I feel like this tool won't be particularly useful without knowing which direction the entrance to the station is, especially given how dangerous the ice caves can be
> Maybe add a small pinpointer or other device suited for this purpose as well?

Not a huge fan of this idea for a few reasons: 
A. I’m trying to keep this as barebones as possible 
B. I am trying to encourage journeys through the caves. No fun if you can just make a b line! 
C. Nanotrasen isn’t willing to spend that much

**Comment by MrDoomBringer** [comment]
i guess it's better than certain death, sure

**Comment by JohnFulpWillard** [comment]
> > Would it be possible to just put a survival pen (the mining one from bepis) in their box, if the map is set to icebox?
> 
> I don’t see how this would solve the issue of people falling into holes with no way out.
>
Because it's a pickaxe in a pen, it's really slow but it still mines

**Comment by Tlaltecuhtli** [comment]
why not make snow diggable by hand? its just snow

**Comment by Pugie** [comment]
> why not make snow diggable by hand? its just snow

it's yellow snow

File: code/modules/mining/equipment/mining_tools.dm (hunk 4 above)
Comment (Cyberboss) [reply code/modules/mining/equipment/mining_tools.dm:120]: It's used intentionally throughout the codebase for better or for worse

**Comment by wesoda25** [comment]
I didn’t know the pen existed, I’ll take a look at that. I have time for this PR now so I’ll see what I can do with the requested changes.

**Comment by wesoda25** [comment]
Yeah that was a lie this isn't gonna happen anytime soon

**Pull request closed without merging**